component/telemetry.go

```diff
@@ -33,6 +33,10 @@ type TelemetrySettings struct {
 	// MeterProvider that the factory can pass to other instrumented third-party libraries.
 	MeterProvider metric.MeterProvider
 
+	// UseOtelForMetrics controls whether the collector should use otel go sdk for
+	// internal telemetry metrics.
+	UseOtelForMetrics bool
```

Comment:
Why exposing this?

Reply:
I've moved it to the internal package `obsreportconfig` so we can use it to the collector core component with otel.